internal/net/http/client/option_test.go

```diff
@@ -22,1945 +22,1378 @@ import (
 	}
 }
 
 func TestWithIdleConnTimeout(t *testing.T) {
-	// Change interface type to the type of object you are testing
-	type T = interface{}
+	type T = transport
 	type args struct {
 		dur string
 	}
 	type want struct {
 		obj *T
-		// Uncomment this line if the option returns an error, otherwise delete it
-		// err error
+		err error
 	}
 	type test struct {
-		name string
-		args args
-		want want
-		// Use the first line if the option returns an error. otherwise use the second line
-		// checkFunc  func(want, *T, error) error
-		// checkFunc  func(want, *T) error
+		name       string
+		args       args
+		want       want
+		checkFunc  func(want, *T, error) error
 		beforeFunc func(args)
 		afterFunc  func(args)
 	}
 
-	// Uncomment this block if the option returns an error, otherwise delete it
-	/*
-	   defaultCheckFunc := func(w want, obj *T, err error) error {
-	       if !errors.Is(err, w.err) {
-	           return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
-	       }
-	       if !reflect.DeepEqual(obj, w.obj) {
-	           return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", obj, w.obj)
-	       }
-	       return nil
-	   }
-	*/
-
-	// Uncomment this block if the option do not returns an error, otherwise delete it
-	/*
-	   defaultCheckFunc := func(w want, obj *T) error {
-	       if !reflect.DeepEqual(obj, w.obj) {
-	           return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", obj, w.obj)
-	       }
-	       return nil
-	   }
-	*/
+	defaultCheckFunc := func(w want, obj *T, err error) error {
+		if !errors.Is(err, w.err) {
+			return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
+		}
+		if diff := comparator.Diff(obj, w.obj, transportComparator...); diff != "" {
+			return errors.New(diff)
+		}
+		return nil
+	}
 
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       args: args {
-		           dur: "",
-		       },
-		       want: want {
-		           obj: new(T),
-		       },
-		   },
-		*/
-
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           args: args {
-		           dur: "",
-		           },
-		           want: want {
-		               obj: new(T),
-		           },
-		       }
-		   }(),
-		*/
+		{
+			name: "set timeout success",
+			args: args{
+				dur: "5s",
+			},
+			want: want{
+				obj: &T{
+					Transport: &http.Transport{
+						IdleConnTimeout: 5 * time.Second,
+					},
+				},
+			},
+		},
+		{
+			name: "set timeout failed with invalid value",
+			args: args{
+				dur: "dummy",
+			},
+			want: want{
+				obj: &T{
+					Transport: &http.Transport{},
+				},
+				err: errors.NewErrCriticalOption("idleConnTimeout", "dummy", errors.New("invalid timeout value: dummy\t:timeout parse error out put failed: time: invalid duration \"dummy\"")),
+			},
+		},
+		{
+			name: "set timeout failed with empty value",
+			args: args{
+				dur: "",
```

Comment:
I think it is not needed.

Reply:
@vankichi Do you mean this line `dur: ""` is not need?
I think it is more clear to write even if the value is the default value of the type, what do you think?

Reply:
but, as you know, the default value of string is `""`.
I do not care but it is not necessary, also your opinion is right.
It is the decision problem, any test rule in coding guideline?

Reply:
There are no such rule in the coding guideline.
I am not sure if I am correct, as all we know the default value of string is "" in golang, we can decide to write this definition or not by ourself (not standardize for everyone, but preference for each developer), what do you think?

Even if we decide to use this pattern, I think this rule only applies to test code, so I think it is better not to write it in coding guideline.

Reply:
I think it is clear creating test case with no input test case like as `int` pattern.
And, I think we consider about the unit test practice and create new document for unit testing.
How about it?

Reply:
> I think it is clear creating test case with no input test case like as int pattern.
I think users need to input the value in option (for example `WithXXX("")`, so I think writing `""` is more realistic and easier to understand.

>  I think we consider about the unit test practice and create new document for unit testing.
I think both ok for me. If you think the code quality of test code is important then I think maybe it is better to create it

Reply:
well, so how about write the default value expressly in each test?

OK. Let's try it, but Not now